### PR TITLE
ECDC-2552: proposal changes all clients

### PR DESCRIPTION
**Pull request by kmurica**

### CODE REVIEW

**Comment by mattclarke** [comment]
Triggering of a change should be done on the server side not the client.
Clients do not know about each other.
There is an event `self.client.experiment.connect` that currently fires when the experiment is finished or the proposal number changes - this might be used for this

File: nicos_ess/gui/panels/exp_panel.py [hunk 1]
```diff
@@ -174,6 +173,7 @@ def initialise_connection_status_listeners(self):
         self.client.connected.connect(self.on_client_connected)
         self.client.disconnected.connect(self.on_client_disconnected)
         self.client.experiment.connect(self.on_experiment_finished)
+        self.client.experiment.connect(self._update_panel)
```
Comment (mattclarke) [inline nicos_ess/gui/panels/exp_panel.py:176]: Isn't this line just the same as the line above it except it calls a different method?
And doesn't `on_experiment_finished` ultimately call `self._update_panel` anyway.
I am struggling to see the point of this.

**Comment by mattclarke** [comment]
Okay, I've looked at the ticket again and I can see it wasn't as clear as it should be.
Currently on main if the proposal number changes all clients update, but if one then subsequently edits the title (or samples, etc.) that change doesn't propagate to all clients.
That is what the ticket is about.

**Comment by ebadkamil** [comment]
> I think, this is not the right way. Normally the title and users of the expriment device are paramters and the parameters are updated via the cache all clients will be informed about the change. Your client has to register on the Exp/title and Exp/users key on then cache and then it should work.

Thanks @jkrueger1 for the suggestion, this indeed is the better approach.

**Comment by kmurica** [comment]
Moved to Gerrit

**Pull request closed without merging**